lib/Lemplate.pm

```diff
@@ -538,6 +534,39 @@ local function stash_set(stash, k, v, default)
     end
 end
 
+_M.vmethods = {
+    join = function(list, delim)
+        if delim == nil then
+            delim = ' '
+        end
+        local out = {}
+        for i = 1, #list, 1 do
+            table.insert(out, list[i])
+            if i ~= #list then
+                table.insert(out, delim)
+            end
+        end
+        return table.concat(out)
```

Comment:
We should avoid such table lookups on hot code paths. Better cache the result of the table lookup of the field `concat` in table `table` in the outer scope. Please fix other similar places.